src/__tests__/Json_decode_test.ml

```diff
@@ -70,10 +80,11 @@ let () =
           expect @@ int (Encode.int big_int) |> toEqual big_int);
       test "infinity" (fun () ->
           let inf = [%raw "Infinity"] in
-          try
-            let (_ : int) = int (Encode.int inf) in
-            fail "should throw"
-          with Decode.DecodeError Json_error "Expected integer, got null" -> pass);
+          expect
+          @@ wrap_exn (fun () ->
+                 let (_ : int) = int (Encode.int inf) in
+                 fail "should throw")
+          |> toEqual "expected an integer but got inf");
```

Comment:
@EmileTrotignon do you know how the error handling can print the OCaml expression name, rather than the JavaScript runtime value?

Reply:
The javascript runtime value likely comes from Console.log, you need a custom printer to do something else, but I have a feeling this wasnt your question. Can you clarify ?

Reply:
I figured it out. Turns out this code returns the string `inf`:

```ml
let infinity = [%raw "Infinity"]

let () = Js.log(Stdlib.string_of_float(infinity))
```